### DISCUSSION TITLE
What happens when Protractor goes away?

### DISCUSSION THREAD

**Discussion by kahan002** [hetznercloud/protractor-test-helper#80] [General]
I just read that Protractor is deprecated as of end of 2022, tentatively (see [https://github.com/angular/protractor/issues/5502](url)). I found these routines helpful, and I am wondering if anyone has given thought to porting them to sit atop something else (maybe directly atop Selenium in Java or Python) or maybe to be rewritten totally to work with Cypress or something else. Is anyone thinking about it, or is this all just going to be deprecated along with Protractor? Best, Jeremy Kahan

**Comment by wachri** [comment]
Thanks for your feedback :).
Yes we also came across this deprecation. Currently we are in an early stage of investigation and no final decision is made. But as we have a huge set of e2e tests the chance is big that we port it in one or another direction to make the shift more easier. One option we have a closer look is definitively Cypress. Hope this helps.

**Reply by kahan002** [reply]
Yes, thank you.

**Comment by wachri** [comment]
Short update here: We already started porting our tests to cypress. I can't say when we will release a port of this lib here or promise something. Currently we are struggling a bit with https://github.com/cypress-io/cypress/issues/7306
